### PR TITLE
Improvement

### LINKED ISSUE #2: Some API updates
1. https://github.com/IBM/MAX-Nucleus-Segmenter/blob/3feedf540d86c9ba818497281b33e99c933d598c/api/predict.py#L10

    Let's rather say "An image file (JPG/PNG/TIFF). Valid input sizes: 64x64, 128x128, 256x256" (assuming we are handling these 3 image formats)

1. https://github.com/IBM/MAX-Nucleus-Segmenter/blob/3feedf540d86c9ba818497281b33e99c933d598c/api/predict.py#L17

    I prefer `probability` to `score`, since it is more in line with standards of other models. Also the help says "Predicted probability for the class label" but should probably be "Predicted probability for presence of the nucleus"

### LINKED ISSUE #3: Add test cases
https://github.com/IBM/MAX-Nucleus-Segmenter/blob/3feedf540d86c9ba818497281b33e99c933d598c/tests/test.py#L32

We need to add a test:
1. for JPG image
2. for non-nucleus image to confirm it returns code `200` and an empty response:

```
{
  "status": "ok",
  "predictions": []
}
```

### LINKED ISSUE #5: README updates
In section https://github.com/IBM/MAX-Nucleus-Segmenter#3-use-the-model, "get predicted labels for the image from the API." -> "get predicted probabilities and segmentation masks for the image from the API."

### LINKED ISSUE #6: Update some model metadata
https://github.com/IBM/MAX-Nucleus-Segmenter/blob/3feedf540d86c9ba818497281b33e99c933d598c/core/model.py#L19

* `id`: `max-nucleus-segmenter`
* `name`: `MAX Nucleus Segmenter`
* `description`: "Nucleus image segmentation model trained with Keras on 2018 Data Science Bowl dataset"
* `type`: "Object Detection" 

(per our standard types, Object Detection includes segmentation models also

### PR DESCRIPTION
**Pull request by feihugis**
This PR addresses the following issues:
- Update some APIs (https://github.com/IBM/MAX-Nucleus-Segmenter/commit/fdf9e7895acc1325869f9b14b4fa7bb6197fb794)
- Add test cases (https://github.com/IBM/MAX-Nucleus-Segmenter/commit/58d6c18b2deb3bab431844545c3be71a03cada2f)
- README updates (https://github.com/IBM/MAX-Nucleus-Segmenter/commit/a88f3a1aba854e08e968b0c2d865cbb67f139fbb)
- Update some model metadata (https://github.com/IBM/MAX-Nucleus-Segmenter/commit/a873669049ab44ccebb70666fad6790376f3c740)

Close #2, close #3, close #5, close #6  

@MLnick

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 # API Metadata
 API_TITLE = 'MAX Nucleus Segmenter'
-API_DESC = 'Detect the nuclei from a microscopy image'
+API_DESC = 'Identify nuclei in a microscopy image, additionally assigning each pixel of the image to a particular nucleus'
```
Comment (bdwyer2) [inline config.py]:
```suggestion
API_DESC = 'Identify nuclei in a microscopy image and assign each pixel of the image to a particular nucleus'
```

File: README.md [hunk 2]
```diff
@@ -3,17 +3,16 @@
 
 # IBM Developer Model Asset Exchange: Nucleus Segmenter
 
-The Nucleus Segmenter model detects the nuclei in the microscopy image. The 
-model is developed based on the architecture of Mask R-CNN using Feature Pyramid 
-network (FPN) and a ResNet50 backbone. Given an image (64 * 64, 128 * 128, or 
-256 * 256), this model outputs the masks and possibilities for each detected 
-nucleus. The mask is compressed using [Run-length encoding (RLE)](https://en.wikipedia.org/wiki/Run-length_encoding).       
+The Nucleus Segmenter model detects nuclei in a microscopy image and additionally specifies the pixels in the image that
```
Comment (MLnick) [inline README.md]: Perhaps remove "additionally" here (similar to Brendan's comment elsewhere)

File: README.md [hunk 3]
```diff
@@ -3,17 +3,16 @@
...
 
-The Nucleus Segmenter model detects the nuclei in the microscopy image. The 
-model is developed based on the architecture of Mask R-CNN using Feature Pyramid 
-network (FPN) and a ResNet50 backbone. Given an image (64 * 64, 128 * 128, or 
-256 * 256), this model outputs the masks and possibilities for each detected 
-nucleus. The mask is compressed using [Run-length encoding (RLE)](https://en.wikipedia.org/wiki/Run-length_encoding).       
+The Nucleus Segmenter model detects nuclei in a microscopy image and additionally specifies the pixels in the image that 
+are assigned to each nucleus. The model is developed based on the architecture of Mask R-CNN using Feature Pyramid 
+network (FPN) and a ResNet50 backbone. Given an image (of size 64 x 64, 128 x 128 or 256 x 256), this model outputs the 
+segmentation masks and probabilities for each detected nucleus The mask is compressed using
```
Comment (MLnick) [inline README.md]: Missing `.` after "detected nucleus"

File: tests/test.py [hunk 4]
```diff
@@ -23,30 +23,45 @@ def test_metadata():
...
 
 def test_predict():
 
     model_endpoint = 'http://localhost:5000/model/predict'
 
-    # Test by the image with multiple nucleis
-    img1_path = 'assets/example.png'
+    # Test by the image with multiple nuclei
+    img_png = 'assets/example.png'
+    img_jpg = 'assets/example.jpg'
```
Comment (MLnick) [inline tests/test.py]: We can move the `jpg` and `tiff` images to the `tests` folder as they are only used for testing. We don't need to include 3 of the same images in `assets`

File: README.md [hunk 5]
```diff
@@ -130,41 +132,47 @@ You should see a JSON response like that below:
...
     {
       "mask": [
-        400,
-        2,
-        463,
-        3,
-        527,
-        3,
-        590,
+        3507,
```
Comment (MLnick) [inline README.md]: Just a question - why has the output for `mask` here been changed?

Comment (feihugis) [reply README.md]: I did not find the image I used at the very beginning for generating the mask output, so I use another image this time. The `example.png` in `Assets` has a very long output and is only used for the demo notebook.

**Comment by feihugis** [comment]
@MLnick The README and test images are updated (https://github.com/IBM/MAX-Nucleus-Segmenter/pull/7/commits/f0551b3981ad918a71cb3eebae5668ab3396a58b).

**Pull request merged**